### DISCUSSION TITLE
How to Manually Modify Drive End Point

### DISCUSSION THREAD

**Discussion by tuxforensics** [teslamate-org/teslamate#2721] [Q&A]
Ok this is just my OCD.. my wife accidently shut down my Banana Pi that was running TeslaMate during the day so my map looks like I flew home line of site.  

I would like to modify the last know drive of that days end point to make it my home.. or do I need to delete the all the drives that day or just leave it. 

Update:  I have since deleted a few drives thinking that it would have cleaned up the visited map but the line from Oshkosh to home is still there.. UGH.. I guess visited is another table that is built.. 

-Mike

** to prevent this, I have since moved my container to a VM in a fiber connected data center.. ( backup and restore of the teslamate db worked great. ) 
<img width="449" alt="Screen Shot 2022-06-26 at 8 53 16 PM" src="https://user-images.githubusercontent.com/80338232/175846346-afd92847-73f3-4ca9-bd58-6773b878a0b5.png">

**Comment by cwanja** [comment]
https://github.com/adriankumpf/teslamate/discussions/2699#discussioncomment-3006475

I just did something similar, but it was deleting some 'rogue' positions. You could drop the drive all together, if you did not care about losing the data. Or update the positions ID for that drive ID - though this seems time consuming.

**Reply by tuxforensics** [reply]
I don't care if I lose the drive but seems like I might have to drop a bunch of them as it can't figure out how I got home but when I turned on the Server it connected my last endpoint and my current location creating a straight line.

**Reply by tobiasehlert** [reply]
There is no way to remove the straight lines straight away. It lines are generated by the `positions` table by lining up all positions in time.

Only option would be to remove the entries in the positions table when this has happened, but you would require to remove as much positions as to the point the car has been at the same GPS locations. Else you will just get another straight line between those to times the car was at different locations.

I have run TeslaMate for multiple years and those type of lines happen from time to time. It could be connection issues where TeslaMate doesn't get data from Tesla properly or even when the car connection drops due to coverage or stuff like that.

Here is how it looks on my TeslaMate installation.. and your will probably look the same in future too:
![image](https://user-images.githubusercontent.com/4417850/176466287-a7107258-652e-49f5-aa0b-889f5ebd9101.png)

**Reply by cwanja** [reply]
> I don't care if I lose the drive but seems like I might have to drop a bunch of them as it can't figure out how I got home but when I turned on the Server it connected my last endpoint and my current location creating a straight line.

If you are OK with losing the data, you can use the trip dashboard to identify what drives constructed the line. And then go dropping those drives. Although, as @tobiasehlert points out, you might need to drop the records from the `positions` table as well. Which would be easy, once you figure out what drive IDs you are going to drop. 

Or, again as Tobias points out, just let it be. I, like you, was a little OCD and corrected my data points. However, it was only about eight or nine records as the GPS corrected itself and removed me from driving in the ocean.

**Reply by tuxforensics** [reply]
Thank you for the info.. I did drop a few from the drives table not the trips.. I guess it doesn't matter.. what happend was my wife shutoff the server by accident during the day.. I now have it in a data center on fiber and backup power.. should be good to go.. Thank you for your input.

**Reply by tuxforensics** [reply]
> There is no way to remove the straight lines straight away. It lines are generated by the `positions` table by lining up all positions in time.
> 
> Only option would be to remove the entries in the positions table when this has happened, but you would require to remove as much positions as to the point the car has been at the same GPS locations. Else you will just get another straight line between those to times the car was at different locations.
> 
> I have run TeslaMate for multiple years and those type of lines happen from time to time. It could be connection issues where TeslaMate doesn't get data from Tesla properly or even when the car connection drops due to coverage or stuff like that.
> 
> Here is how it looks on my TeslaMate installation.. and your will probably look the same in future too: ![image](https://user-images.githubusercontent.com/4417850/176466287-a7107258-652e-49f5-aa0b-889f5ebd9101.png)

Thank you.. I will just live with it at this point..

**Reply by tobiasehlert** [reply]
> Thank you for the info.. I did drop a few from the drives table not the trips.. I guess it doesn't matter.. what happend was my wife shutoff the server by accident during the day.. I now have it in a data center on fiber and backup power.. should be good to go.. Thank you for your input.

Well then you got two options.. switch to a datacenter or switch wife 😆

No I'm kidding.. I had two young boys that love electrical outlets. I have modified furniture at home preventing outages like this, where the server go off, but there still occur things that can happen. 😄
